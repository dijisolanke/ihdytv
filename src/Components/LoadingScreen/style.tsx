import styled, { keyframes } from "styled-components";

// Keyframe animation for fading in the text
const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`;

// Style for the overlay (full screen and centered)
export const OverlayStyle = styled.div`
  position: "fixed";
  top: 0;
  left: 0;
  width: "100%";
  height: "100%";
  background-color: "white";
  display: "flex";
  justify-content: "center";
  align-items: "center";
  font-size: "2rem";
  color: "black";
  z-index: 9999;
`;

// Loading circle with dynamic background based on progress
export const LoadingCircle = styled.div<{ progress: number }>`
  position: absolute;
  top: 43.6vh;
  right: 46.2%;
  width: 100px;
  height: 100px;
  border-radius: 50%;
  background: ${({ progress }) =>
    `conic-gradient(#9c9999 ${progress}%, #ddd ${progress}%)`};
  transition: background 0.1s ease;
`;

// Style for the text that fades in
export const LoadingText = styled.span`
  position: absolute;
  top: 50vh;
  left: 50%;
  transform: translate(-50%, -50%);
  font-size: 1.5rem;
  color: black;
  z-index: 1;
  opacity: 0;
  animation: ${fadeIn} 1.5s ease forwards;
`;
