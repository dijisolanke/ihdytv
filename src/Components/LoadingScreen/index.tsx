// LoadingOverlay.js

import React, { useState, useEffect } from "react";
import { OverlayStyle, LoadingText, LoadingCircle } from "./style";

interface LoadingOverlayProps {
  progress: number; // Explicitly type progress as a number
}

// Inline style for text with opacity transition
const textStyle: React.CSSProperties = {
  position: "absolute",
  top: "50vh",
  left: "50%",
  transform: "translate(-50%, -50%)",
  fontSize: "1.5rem",
  color: "black",
  zIndex: 1,
  opacity: 0,
  transition: "opacity 1.5s ease",
};

// LoadingOverlay component
const LoadingScreen = ({ progress }: LoadingOverlayProps) => {
  const [fadeIn, setFadeIn] = useState(false);

  useEffect(() => {
    setFadeIn(true); // Trigger fade-in effect
  }, []);

  return (
    <OverlayStyle>
      <div style={{ position: "relative" }}>
        <LoadingCircle progress={progress} />
        <LoadingText style={{ opacity: fadeIn ? 1 : 0 }}>IHDY</LoadingText>
      </div>
    </OverlayStyle>
  );
};

export default LoadingScreen;
