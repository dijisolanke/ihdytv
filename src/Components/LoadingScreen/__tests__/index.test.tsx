import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
import LoadingScreen from ".."; // Adjust the path based on your file structure

jest.mock("../style", () => ({
  OverlayStyle: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="overlay-style">{children}</div>
  ),
  LoadingText: (props: React.HTMLAttributes<HTMLDivElement>) => (
    <div data-testid="loading-text" {...props} />
  ),
  LoadingCircle: ({ progress }: { progress: number }) => (
    <div data-testid="loading-circle" data-progress={progress} />
  ),
}));

describe("LoadingScreen", () => {
  it("renders the overlay container and its child components", () => {
    // Arrange
    const progress = 50;

    // Act
    render(<LoadingScreen progress={progress} />);

    // Assert
    expect(screen.getByTestId("overlay-style")).toBeInTheDocument();
    expect(screen.getByTestId("loading-circle")).toBeInTheDocument();
    expect(screen.getByTestId("loading-circle")).toHaveAttribute(
      "data-progress",
      progress.toString()
    );
    expect(screen.getByTestId("loading-text")).toBeInTheDocument();
  });

  it("matches the snapshot for consistent UI", () => {
    // Arrange
    const progress = 90;

    // Act
    const { asFragment } = render(<LoadingScreen progress={progress} />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
