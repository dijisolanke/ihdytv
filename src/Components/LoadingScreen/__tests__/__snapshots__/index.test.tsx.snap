// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen matches the snapshot for consistent UI 1`] = `
<DocumentFragment>
  <div
    data-testid="overlay-style"
  >
    <div
      style="position: relative;"
    >
      <div
        data-progress="90"
        data-testid="loading-circle"
      />
      <div
        data-testid="loading-text"
        style="opacity: 1;"
      >
        IHDY
      </div>
    </div>
  </div>
</DocumentFragment>
`;
